File: Phoenix/lib/biab_starter/biab/resources/marketing.ex
defmodule BiabStarter.Biab.Resources.Marketing do
  @moduledoc """
  Schema-driven marketing content.

  `biab.config.ts` declares the SHAPE (`npm run sync-schema` pushes it), the
  dashboard fills in the CONTENT, and `page_bundle/3` reads it back. Every
  section should render a local fallback when the bundle is missing.
  """

  alias BiabStarter.Biab.Client

  def page_bundle(client, page_key \\ "home", locale \\ nil),
    do: Client.get(client, Client.site_path(client, "marketing/bundle"), pageKey: page_key, locale: locale)

  def published_schema(client),
    do: Client.get(client, Client.site_path(client, "marketing/published-schema"))

  def locales(client), do: Client.get(client, Client.site_path(client, "marketing/locales"))
  def branding(client), do: Client.get(client, Client.site_path(client, "branding"))
end

File: Laravel/app/Biab/Resources/Marketing.php
<?php

namespace App\Biab\Resources;

use App\Biab\Client;

/**
 * Schema-driven marketing content.
 *
 * `biab.config.ts` declares the SHAPE (run `npm run sync-schema` to push it);
 * the dashboard fills in the CONTENT; `getPageBundle()` reads it back. Every
 * section here should render a local fallback when the bundle is missing, so
 * a fresh clone with no BIAB env still shows a complete page.
 */
class Marketing
{
    public function __construct(private readonly Client $client)
    {
    }

    /** @return array<string, mixed> */
    public function getPageBundle(string $pageKey = 'home', ?string $locale = null): array
    {
        return $this->client->get($this->client->sitePath('marketing/bundle'), [
            'pageKey' => $pageKey,
            'locale' => $locale,
        ]);
    }

    /** @return array<string, mixed> */
    public function getPublishedSchema(): array
    {
        return $this->client->get($this->client->sitePath('marketing/published-schema'));
    }

    /** @return array<string, mixed> */
    public function getLocales(): array
    {
        return $this->client->get($this->client->sitePath('marketing/locales'));
    }

    /** @return array<string, mixed> */
    public function branding(): array
    {
        return $this->client->get($this->client->sitePath('branding'));
    }
}
